src/worker/queue.ts

```diff
@@ -48,3 +59,83 @@ export function startQueue(
 
     return worker
 }
+
+function startQueueKafka(
+    server: PluginsServer,
+    processEvent: (event: PluginEvent) => Promise<PluginEvent | null>
+): Queue {
+    const eventsProcessor = new EventsProcessor(server)
+
+    eventsProcessor.kafkaConsumer.on('ready', () => {
+        eventsProcessor.kafkaConsumer.subscribe([KAFKA_EVENTS_WAL])
+        // consume event messages in batches of 1000 every 50 ms
+        eventsProcessor.consumptionInterval = setInterval(() => {
+            eventsProcessor.kafkaConsumer.consume(
+                1000,
+                async (error: LibrdKafkaError, messages: Message[]): Promise<void> => {
+                    if (error) {
+                        console.error('⚠️ Error while consuming!')
+                        console.error(error)
+                        Sentry.captureException(error)
+                    }
+                    if (messages?.length) {
+                        console.info(
+                            `🍕 ${messages.length} ${
+                                messages.length === 1 ? 'message' : 'messages'
+                            } consumed from Kafka at ${Date.now()}`
+                        )
+                    } else {
+                        return
+                    }
+                    try {
+                        for (const message of messages) {
+                            const timer = new Date()
+                            const rawEventMessage = JSON.parse(message.value!.toString()) as RawEventMessage
+                            const parsedEventMessage: ParsedEventMessage = {
+                                ...rawEventMessage,
+                                data: JSON.parse(rawEventMessage.data),
+                                now: DateTime.fromISO(rawEventMessage.now),
+                                sent_at: rawEventMessage.sent_at ? DateTime.fromISO(rawEventMessage.sent_at) : null,
+                            }
+                            console.info(`Processing event ${parsedEventMessage.data.event} from WAL`)
+                            const processedEvent = await processEvent({
```

Comment:
This bit should run `processEventBatch` on all the events at once.

Reply:
It definitely should, though I don't think we run it anywhere inside the Celery pipeline at all yet? I'll give it a go here though.

Reply:
Yeah, we don't use it for Celery as celery only returns events one by one. The `processEventBatch` function was basically made with Kafka in mind. It might be wise to implement some batching in front of celery as well, to for example batch all events every 100ms, up to 100 per batch, but that's for another day.